### PR TITLE
Adds shiny search to the plugin install screen, also debounce search handling on plugin screen

### PR DESCRIPTION
**Pull request by adamsilverstein**

### CODE REVIEW

File: js/shiny-updates.js [hunk 1]
```diff
@@ -1075,7 +1075,37 @@ window.wp = window.wp || {};
...
 		} );
 
-		$( '#plugin-search-input' ).on( 'keyup search', function() {
+		/**
+		 * Handle changes to the plugin search box on the new-plugin page, searching the repository dynamically.
+		 *
+		 * @todo Add a spinner during search?
+		 */
+		$( 'input.wp-filter-search' ).on( 'keyup search', _.debounce( function() {
+			var val  = $( this ).val(),
```
Comment (obenland) [inline js/shiny-updates.js]: We could just assign it to `'s'` inline, it's not used anywhere else, is it?

File: js/shiny-updates.js [hunk 2]
```diff
@@ -1075,7 +1075,37 @@ window.wp = window.wp || {};
...
+				data = {
+					'_ajax_nonce': wp.updates.ajaxNonce,
+					's':           val,
+					'tab':         'search'
+				},
+				jqxhr;
```
Comment (obenland) [inline js/shiny-updates.js]: I think we can remove that, it doesn't look like it's used. Did JSlint not pick that up?

File: shiny-updates.php [hunk 3]
```diff
@@ -628,3 +629,35 @@ function wp_ajax_search_plugins() {
...
+	check_ajax_referer( 'updates' );
+
+	global $wp_list_table, $hook_suffix;
+	$hook_suffix = 'plugins.php';
+
+	$status        = array();
+	$wp_list_table = _get_list_table( 'WP_Plugin_Install_List_Table' );
+
+	if ( ! $wp_list_table->ajax_user_can() ) {
+		$status['error'] = __( 'You do not have sufficient permissions to manage plugins on this site.' );
```
Comment (obenland) [inline shiny-updates.php]: We never handle that error, do we? I wonder if we should either just fail silently here or display that error somehow to the user.

**Comment by obenland** [comment]
Is there a way we can make this more testable?

File: shiny-updates.php (hunk 3 above)
Comment (adamsilverstein) [inline shiny-updates.php]: good point, we should probably return and display the error to the user.

File: js/shiny-updates.js (hunk 2 above)
Comment (adamsilverstein) [inline js/shiny-updates.js]: removed. jscs picked it up after i merged master back into my branch,

File: js/shiny-updates.js (hunk 1 above)
Comment (adamsilverstein) [inline js/shiny-updates.js]: Good point. I was matching what we have below in the search for #plugin-search-input, I will change both of those.

**Comment by adamsilverstein** [comment]
re: testability, thats a good question, I don't really know how. We should be able to test the ajax endpoints. I'd love more tests around these features.

**Pull request merged**